### PR TITLE
Fix the master-next build

### PR DESCRIPTION
**Pull request by MForster**
See https://github.com/apple/swift/pull/31118#discussion_r413975500 for context.

@adrian-prantl @gribozavr

### CODE REVIEW

**Comment by gribozavr** [comment]
@swift-ci Please smoke test

**Comment by gribozavr** [comment]
CI failures are in LLDB. This PR is fixing the Swift compiler build, which succeeded in CI, hence I'm merging this PR.

**Review by gribozavr — APPROVED** [review]

File: lib/ClangImporter/ClangAdapter.cpp [hunk 1]
```diff
@@ -81,7 +81,7 @@ const clang::Decl *
 importer::getFirstNonLocalDecl(const clang::Decl *D) {
   D = D->getCanonicalDecl();
   auto iter = llvm::find_if(D->redecls(), [](const clang::Decl *next) -> bool {
-    return !next->isLexicallyWithinFunctionOrMethod();
+    return !next->isInLocalScope();
```
Comment (adrian-prantl) [inline lib/ClangImporter/ClangAdapter.cpp:84]: This one isn't particularly obvious; I assume that is how the upstream Clang sources changed, too?

**Comment by adrian-prantl** [comment]
Thanks!

File: lib/ClangImporter/ClangAdapter.cpp (hunk 1 above)
Comment (MForster) [reply lib/ClangImporter/ClangAdapter.cpp:84]: Here is the change: https://github.com/apple/llvm-project/commit/f43859a099fa3587123717be941fa63ba8d0d4f2

`isInLocalScope` is the new name of the method. The implementation did change, though.

**Pull request merged**